### ISSUE TITLE
textTracks crash Video on physical device iOS (6.0.0@alpha4)

### ISSUE THREAD

**Issue by dvijeniii05** [react-native-community/react-native-video#2965]
# Bug

Adding textTracks and selectedTextTracks to Video component breaks the Video on physical device (iOS only). Running on emulator or XCode + device doesn't result in the issue and displays video and subtitles as desired.
Not including textTracks and selectedTextTrack doesn't cause any issues and video works as expected.
Similar issue was raised before and meant to have a fix in @alpha4 version, but checked it today and the issue is still present.
![photo_2022-12-20 18 53 50](https://user-images.githubusercontent.com/67027576/208744177-62bf8d49-7830-4006-a8ae-1b9699cd46eb.jpeg)

## Platform

iOS - only physical device

Which player are you experiencing the problem on:
* iOS

## Environment info

"react-native-video": "6.0.0-alpha.4"
"react-native": "0.69.5",

## Steps To Reproduce

Create a simple Video component.
Use a .mp4 typed video to pass into source.
Pass in .VTT typed subtitles into textTracks props.
Use selectedTextTrack prop.
Launch any app with this Video component on a physical device (iOS)

## Expected behaviour

Should display video and subtitles.

## Reproducible sample code

The issues is present only on physical device (iOS only)

```
<Video
        ignoreSilentSwitch="ignore"
        poster={imageUrl}
        controls={true}
        posterResizeMode="contain"
        source={{
          uri: sourceFile,
        }}
        textTracks={[
          {
            title: 'English CC',
            language: 'en',
            type: TextTrackType.VTT,
            uri: 'https://bitdash-a.akamaihd.net/content/sintel/subtitles/subtitles_en.vtt',
          },
        ]}
        selectedTextTrack={{
          type: 'language',
          value: 'en',
        }}
      />
```

Any help or advice is much appreciated!

**Comment by cihanbas** [comment]
i'm getting same issue

**Comment by dvijeniii05** [comment]
We have managed to get extra logging using 'Console.app' and physical device connected through the usb. The log that looks strange and only appears when running on physical device: 
```
“ mediaserverd <<<< FigFilePlayer >>>> playerfig_prepareWorkingItem2: (0x7a36f1a00:0x7a3097c00) P/VI I/MAZ.01 posting FailedToBecomeReadyForPlayback due to error -12783 ” 
```

We also received extra logs through onError: 

```
{"error" : {
"localizedRecoverySuggestion":"",
"domain":"AVFoundationErrorDomain",
"localizedDescription":"The operation could not be completed",
"localizedFailureReason":"An unknown error occured (-12783)",
"code":-11800
},
"target":582
}
```

 However after checking on dev website this error code (-11800) stands for “Unknown error”.

We have also tried using different variations of props and these are the results: 

Using only textTrack prop and not defining selectedTextTrack props results in NO error but doesn’t display subtitles (as expected)

Using different type in selectedTextTracks such as index for example, still results in an error.

**Comment by sam1463** [comment]
This is a duplicate of #2890 isn't it?

**Comment by Alpha4** [comment]
Can you stop mentioning me please ?

**Comment by dvijeniii05** [comment]
@sam1463 potentially.
UPD: we discovered our issue and it is working now.

We deploy our app to TestFlight through ADO and ADO was initially using Xcode 13 to build and distribute the app. This has changed recently and updated by microsoft to Xcode 14. And after this update the video works fine with textracks.

We couldnt replicate the issue locally as we were using XCode 14. 

Also, just to check, we installed XCode 13 to one of our machines, produced a local build through it (with subtitles) and the video was crashing again.

So it was definitely related to Xcode version. Hope this will help someone else as well.
